### ISSUE TITLE
Spurious PE entries in Hcal hits at around 817 PE

### ISSUE THREAD

**Issue by EinarElen** [LDMXAnalysis/ldmx-sw#1344]
As noted by @jpasc27 and @mrsolt, we seem to have some odd entries in the HCal rechits with occasional photoelectrons > what you get at max ADC (see [Jessica's talk at the 2023 spring collaboration meeting ](https://indico.fnal.gov/event/58323/contributions/259798/subcontributions/8885/attachments/166149/221105/LDMX%20CM%20Hcal%20Studies%20Update.pdf) slide 5). 

As far as I understood it, Jessica does not take the `maxPE` hits from the veto processor so this should be unrelated to LDMX-Software/Hcal#58. It also looks like the hits always have the same value, which wouldn't make sense if it was LDMX-Software/Hcal#58. Unclear if it is related to LDMX-Software/ldmx-sw#1348 

I tested this quickly with a neutron gun while working on the HCal DQM today and can confirm that I'm getting the same thing.

![Screenshot from 2023-04-24 18-04-38](https://user-images.githubusercontent.com/14233914/234053226-53884313-c8e6-4962-842e-d86e6ce55f1e.png)

All of the RecHits included in the figure above are from the Back HCal.

**Comment by EinarElen** [comment]
Some details on one of the RecHits in question... 
```C++
HcalHit { id: 18000414,  energy: 55.9885MeV, time: 36.5234ns, amplitude: 0, pe: 817}
```

Hit ID corresponds to section 0, layer 1, strip 20 so nothing looks immediately suspicious here _other than the 0 amplitude_.
``` C++
{<ldmx::HcalAbstractID> = {<ldmx::DetectorID> = {static SUBDETECTORID_MASK = 63, static SUBDETECTORID_SHIFT = 26,
      static SUBDETECTOR_PAYLOAD_MASK = 67108863, id_ = 402654228}, static BAR_TYPE_MASK = 7, static BAR_TYPE_SHIFT = 23, static HCAL_PAYLOAD_MASK = 8388607},
  static SECTION_MASK = 7, static SECTION_SHIFT = 18, static LAYER_MASK = 255, static LAYER_SHIFT = 10, static STRIP_MASK = 255, static STRIP_SHIFT = 0}
  ```
  
  Looking more closely: 
- `isADC` -1700121632 (hmm...)
- `isNoise` false 
- `minpe` 0 (really odd if the other end has 800!)
- `xpos`  -6847.9707 (!!)
- `ypos` 25 
- `zpos` 879 (think that matches the layer above more or less) 

Given the `xpos` value, I now very much think that this is the same issue as LDMX-Software/ldmx-sw#1348. Interesting to see that these entries might have bogus values for `isADC`!

**Comment by EinarElen** [comment]
Last comment can be ignored, `isADC` is always containing junk :(

This does mean that `isTOT` is always false, since it is defined as `!isADC`

**Comment by tvami** [comment]
> I tested this quickly with a neutron gun while working on the HCal DQM today and can confirm that I'm getting the same thing.

@EinarElen can you show what you did to reproduce this? I'm wondering if this issue is gone now with having all the initializations fixed?

**Comment by tvami** [comment]
Maybe this is slightly different then the one above, but this is what I see
<img width="345" height="418" alt="Image" src="https://github.com/user-attachments/assets/f397e232-10d0-4326-85c2-0ab1e49dbe6e" />

I've cornered it to be for the case when we are in ADC mode and the ADC saturates. What do we want to do with this case? Flag the hit to be saturated and move on? Change the definition of the PE? Something else? @bechenard what's your take?

**Comment by EinarElen** [comment]
This is just saturation of the ADC, this isn't the bug we referenced in the original issue, so your plot (if it had maybe 10x statistics) would actually be showing the bug being resolved :) 

The bug isn't the big "peak" at the maximum, its the entry that appears above the saturation value

**Comment by tvami** [comment]
Yes, I understand that this is a feature of the ADC mode, but I wonder if that means we should be in ToT mode in these cases?

**Comment by tvami** [comment]
> (if it had maybe 10x statistics) 

here is the same plot with >1M HCal gun events in logY scale now:

<img width="580" height="625" alt="Image" src="https://github.com/user-attachments/assets/a983562e-7597-4122-b941-f8ac51c37e4f" />

**Comment by tvami** [comment]
Same for the back HCAL:

<img width="578" height="612" alt="Image" src="https://github.com/user-attachments/assets/c765f519-c93e-4fe2-b762-8877fdfe5b81" />
